### ISSUE TITLE
credo consumes a lot of memory until it get's killed after upgrading to 0.7

### ISSUE THREAD

**Issue by manukall** [rrrene/credo#368]
### Precheck

* For proposing a new feature, be sure to name your issue "Proposal: ..."
* For bugs, please do a quick search and make sure the bug has not yet been reported here

### Environment

* Credo version (`mix credo -v`): 0.7.3
* Erlang/Elixir version (`elixir -v`):
```
Erlang/OTP 19 [erts-8.3] [source] [64-bit] [smp:4:4] [async-threads:10] [hipe] [kernel-poll:false]
Elixir 1.4.2
```
* Operating system: Linux

### What were you trying to do?
run `mix credo` on a reasonably large project (`Checking 561 source files (this might take a while)`)

### Expected outcome
Credo should report problems without crashing

### Actual outcome
Credo runs for some time and then fills my memory in a short period. I've put some `IO.inspect` statements into the runner code, and it passes the "run on all checks". after that, the memory usage rises very quickly and the credo process get's killed by the OOM killer.

Version 0.6 works fine, btw. Only upgrading to 0.7 causes the crashes.

I'm not sure how to further debug this. Please let me know what further information I can provide.

**Comment by rrrene** [comment]
Ouch. I will try to reproduce this once I get home from ElixirConf! :+1: Thx for the report!

**Comment by rrrene** [comment]
@manukall First: Thanks for reporting this. It took a while for me to understand the underlying problem, but it should now be resolved on `master`. 

You can check by setting the dep to `{:credo, github: "rrrene/credo"}`. Please report your findings!

**Comment by manukall** [comment]
@rrrene master works for me. thanks!

**Comment by rrrene** [comment]
Great! I released `v0.8.0-rc1` containing the fix and will release `v0.8.0` later this week.
